amoro-ams/src/test/java/org/apache/amoro/server/optimizing/maintainer/TestDataExpire.java

```diff
@@ -591,6 +592,7 @@ protected static Map<String, String> getDefaultProp() {
     prop.put(TableProperties.ENABLE_DATA_EXPIRATION, "true");
     prop.put(TableProperties.DATA_EXPIRATION_FIELD, "op_time");
     prop.put(TableProperties.DATA_EXPIRATION_RETENTION_TIME, "1d");
+    prop.put("write.metadata.metrics.default", "none");
```

Comment:
Does "write.metadata.metrics.default" need to be added to TableProperties ? @XBaith

Reply:
You are right. It should not be the default property. It's a good idea to create a new test case to check that `write.metadata.metrics.default` is `none` to trigger data expiration based on partition informatio

Reply:
You're right, I should create a set of test cases with "write.metadata.metrics.default" set to "none." However, I'm currently stuck with the issue that configuring "write.metadata.metrics.default" as "none" in the test cases doesn't take effect. Even when I set "write.metadata.metrics.default" to "none," the upper_bound is still being recorded in the metrics. Could you help me take a look?

Reply:
During debugging, I found that the relevant configuration for metrics was not correctly passed when writing Parquet files, leading to "none" not taking effect. However, in my production environment, there is no issue.

<img width="1432" alt="image" src="https://github.com/user-attachments/assets/b935ad1b-4640-48ce-9995-ba28576757de" />
<img width="1134" alt="image" src="https://github.com/user-attachments/assets/b82a8843-cb15-4a51-9277-fe4ae1b1c542" />

Reply:
Looks like iceberg GenericAppenderFactory do not put the right metric configs. Currently, I think we can just add a new test for setting metric property as `none`, then I will report the issue and try to fix it in Iceberg.